src/bootstrap/bin/rustc.rs

```diff
@@ -126,6 +129,13 @@ fn main() {
         }
     }
 
+    if is_proc_macro && stage == "0" {
+        // Cargo doesn't pass RUSTFLAGS to proc_macros:
+        // https://github.com/rust-lang/cargo/issues/4423
+        // Set `--cfg=bootstrap` explicitly instead.
+        cmd.arg("--cfg=bootstrap");
+    }
```

Comment:
I'm wondering if we should actually move this to the else branch above; that's the "not a target dep" (so includes proc macros, build scripts, ...other things?) code. Is there a reason we only try to target proc macros here?

Reply:
OMG that's exactly what I needed, that fixes all the bugs. Thank you so much!